packages/astro/src/core/create-vite.ts

```diff
@@ -112,7 +115,13 @@ export async function createVite(
 		customLogger: createViteLogger(logger, settings.config.vite.logLevel),
 		appType: 'custom',
 		optimizeDeps: {
-			entries: ['src/**/*'],
+			// Scan all files within `srcDir` except for known server-code (e.g endpoints)
+			entries: [
+				`${srcDirPattern}!(pages)/**/*`, // All files except for pages
+				`${srcDirPattern}pages/**/!(*.js|*.ts)`, // All pages except for endpoints
```

Comment:
Ehehe, unfortunately `mjs` and `cjs` should be included, even though it's a low chance that users use them. Still, we should

Reply:
Good catch! I'll add this too